### PR TITLE
RavenDB-17292 - add new license properties

### PR DESCRIPTION
**Pull request by grisha-kotler**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17292
https://issues.hibernatingrhinos.com/issue/RavenDB-17294
https://issues.hibernatingrhinos.com/issue/RavenDB-17295

### Additional description

Add new license features

### Type of change

- Low

### Backward compatibility

- Ensured. Old licenses will work with the new licensing schema.

### Testing 

- It has been verified by manual testing

### CODE REVIEW

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/Commercial/LicenseValidator.cs [hunk 1]
```diff
@@ -132,6 +134,13 @@ private static bool TryGetTermIndexAndType(BinaryReader br, out string licensePr
             return true;
         }
 
+        private static string GetString(byte[] bytes)
+        {
+            var chars = new char[bytes.Length / sizeof(char)];
+            Buffer.BlockCopy(bytes, 0, chars, 0, bytes.Length);
+            return new string(chars);
+        }
+
```
Comment (ayende) [inline src/Raven.Server/Commercial/LicenseValidator.cs]: No, use `Encoding.UTF8.GetString()`, instead.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Commercial/LicenseValidator.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Server/Commercial/LicenseValidator.cs]: done

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/Commercial/LicenseStatus.cs [hunk 2]
```diff
@@ -76,11 +51,48 @@ public bool Expired
...
+                    var typeAsInt = (int)type;
+                    if (Enum.IsDefined(typeof(LicenseType), typeAsInt))
+                        return (LicenseType)typeAsInt;
+                }
+
+                return LicenseType.Community;
+            }
+        }
+
+        public double Version => GetValue<int?>("version") / (double)10 ?? -1;
```
Comment (ppekrol) [inline src/Raven.Server/Commercial/LicenseStatus.cs]: why double? easier to handle int, like 40 or 50

Comment (grisha-kotler) [reply src/Raven.Server/Commercial/LicenseStatus.cs]: done

**Pull request merged**